mono/utils/mono-time.c

```diff
@@ -161,6 +178,31 @@ mono_100ns_ticks (void)
 }
 
 /*
+ * Returns user + system time for a thread in 100ns units.
+ */
+gint64
+mono_thread_cpu_time (void)
+{
+#if defined(__linux__)
+	struct timespec ts;
+	clock_gettime (CLOCK_THREAD_CPUTIME_ID, &ts);
+	return ts.tv_sec * 10000000 + ts.tv_nsec / 100;
+#elif defined(PLATFORM_MACOSX)
+	mach_msg_type_number_t count = THREAD_BASIC_INFO_COUNT;
+	thread_basic_info_data_t info;
+
+	if (thread_info (pthread_mach_thread_np (pthread_self ()), THREAD_BASIC_INFO,
```

Comment:
Like pretty much everything we do...

But anyway.  I benchmarked this vs `mach_thread_self()`/`mach_port_deallocate()` on the Roslyn benchmark, which is very parallel and does a lot of compilation.  No noticable difference.  So I'm fine with either.